### PR TITLE
Standardize `CompressionTypeVariant` encoding in protobuf

### LINKED ISSUE #8680: Standardize compression encoding in protobuf
### Is your feature request related to a problem or challenge?

We currently have two mechanisms for serializing compression formats in datafusion.proto.

`CreateExternalTableNode` uses a string representation:

```protobuf
string file_compression_type = 10;
```

`JsonWriterOptions` uses an enum:

```protobuf
CompressionTypeVariant compression = 1;
```

Here is the definition of `CompressionTypeVariant`:

```protobuf
enum CompressionTypeVariant {
  GZIP = 0;
  BZIP2 = 1;
  XZ = 2;
  ZSTD = 3;
  UNCOMPRESSED = 4;
}
```

### Describe the solution you'd like

It would be better to standardize on one approach for a more consistent API.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tushushu**
## Which issue does this PR close?

Closes #8680.

### CODE REVIEW

**Comment by tushushu** [comment]
This PR is not ready for a review yet.

**Review by metesynnada — APPROVED** [review]
LGTM.

File: datafusion/proto/proto/datafusion.proto [hunk 1]
```diff
@@ -212,7 +212,7 @@ message CreateExternalTableNode {
   bool if_not_exists = 7;
   string delimiter = 8;
   string definition = 9;
-  string file_compression_type = 10;
+  CompressionTypeVariant file_compression_type = 10;
```
[the remark this answers is not in the document]
Comment (tushushu) [reply datafusion/proto/proto/datafusion.proto]: I was not aware of this, thanks @jonahgao for the advice.

**Comment by tushushu** [comment]
> I took the liberty if making @jonahgao 's suggestion to use `reserved` as well as resolving the clippy error to get this PR's CI checks green

Thanks so much Andrew for the commits, I was supposed to submit the code yesterday.

**Pull request merged**